### PR TITLE
Added JSON schema and associated unit test

### LINKED ISSUE #72: Author JSON Schema for item database
After some research and testing, it seems that the best way to validate the item database contents (the JSON files) is to use a [JSON schema](https://json-schema.org/). A branch called `json-schema` currently is the work-in-progress towards the schema and an associated pytest module.

### PR DESCRIPTION
**Pull request by osrsbox**
This closes #72 , and adds to #18 by adding additional tests

Basically, this PR adds a unit test for the contents of the JSON files in `docs/items-json`. I have added a JSON schema (`test/item-schema.json`) that provides item properties, associated data types, and required properties. The new test (`test/test_items_database.py`) loads the schema, and checks each JSON file. 

Notes: 

- Had to remove two outdated (nulled) items that are no longer in game
- Had to update `items-complete.json` for this to pass the test (due to the removed items)
- Had to add `jsonschema` package to the requirements

@nwalsh1995 - would really appreciate a review on this PR. Some feedback would be good before merging.

### CODE REVIEW

File: test/test_items_database.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+
+def test_item_database(path_to_docs_dir: Path):
+    """Unit test to check item database contents against JSON schema
+
+    :param path_to_docs_dir: The path to the `docs` folder.
+    """
+    # Read in the item_schema.json file
+    path_to_schema = Path("test/item_schema.json")
+    with open(path_to_schema, 'r') as f:
```
Comment (nwalsh1995) [inline test/test_items_database.py]: Paths from pathlib are pretty cool in the fact that they remove this boiler plate with `.read_text` https://docs.python.org/3.6/library/pathlib.html#pathlib.Path.read_text

**Comment by nwalsh1995** [comment]
As far as validation goes, it seems to me that this is a clean design that does not force the user into specific types for certain attributes which is good since we want to keep Python a dynamic language as much as possible. :+1:

File: test/test_items_database.py (hunk 1 above)
Comment (osrsbox) [reply test/test_items_database.py]: Nice hint, I will change this. The more I use pathlib the more I can see why it is such a good library.

**Comment by osrsbox** [comment]
Thanks for the review @nwalsh1995 - will merge this PR today.

**Pull request merged**